jobs/sql/extractions/all_encounters.sql

```diff
@@ -0,0 +1,108 @@
+SET @partition = '${partitionNum}';
+set sql_safe_updates = 0;
+set @next_appt_date_concept_id = CONCEPT_FROM_MAPPING('PIH', 5096);
+
+drop temporary table if exists temp_all_encounters;
+create temporary table temp_all_encounters
+(
+    encounter_id         int,
+    encounter_datetime   datetime,
+    patient_id           int,
+    visit_id             int,
+    creator              int(11),
+    user_entered         varchar(255),
+    encounter_location   varchar(255),
+    encounter_type_name  varchar(50),
+    entered_datetime     datetime,
+    emr_id               varchar(15),
+    next_appt_date       date,
+    voided               bit,
+    last_updated         datetime(3)
+);
+
+-- If there is not a previous watermark, initialize with all encounters
+insert into temp_all_encounters (encounter_id)
+select encounter_id from encounter
+where @previousWatermark is null;
+
+-- If there is a previous watermark, initialize with only those encounters since that watermark
+insert into temp_all_encounters (encounter_id)
+select encounter_id from encounter
+inner join dbevent_patient dp on encounter.patient_id = dp.patient_id
+where @previousWatermark is not null
+and dp.last_updated >= @previousWatermark
+and dp.last_updated <= @newWatermark;
+
+create index temp_all_encounters_encounterId ON temp_all_encounters (encounter_id);
+
+-- Load the encounters
+update temp_all_encounters t
+inner join encounter e on t.encounter_id = e.encounter_id
+set t.encounter_datetime = e.encounter_datetime,
+    t.encounter_type_name = encounter_type_name_from_id(e.encounter_type),
+    t.encounter_location = location_name(e.location_id),
+    t.patient_id = e.patient_id,
+    t.visit_id = e.visit_id,
+    t.user_entered = person_name_of_user(e.creator),
+    t.entered_datetime = e.date_created,
+    t.voided = e.voided,
+    t.last_updated = @newWatermark
+;
+
+CREATE INDEX temp_all_encounters_patientId ON temp_all_encounters (patient_id);
+
```

Comment:
The above queries are the main things to pay attention to here, and will need to be replicated similarly in any other jobs that we want to use incremental loading on (or some better method we come up with).  Basically, this loads temp_all_encounters with all encounters if there is no previous watermark, but loads it with a subset of encounters (based on which patients have changed in the watermark window) if the watermark dates are not null.

Then I moved the logic for what columns are in this table and how they are defined into a single update statement so that we wouldn't have to define them in 2 places.

Notice that the "last_updated" is populated with the newWatermark value